### PR TITLE
fix: use next-runtime-env for picking up at runtime NEXT_PUBLIC envir…

### LINKED ISSUE #147: 🐞 BUG REPORT
### 🔄 Duplicate Check

- [X] I have searched both open and closed issues and found no similar reports.

### 🔍 What Happened?

When I change the environment variable NEXT_PUBLIC_CKAN_URL, this is only picked up at build time and not at runtime 

### 👣 Steps to Reproduce

1. Run local and go to localhost:3000
2. Change NEXT_PUBLIC_CKAN_URL inside the docker compose to something else 
3. The link with back end is broken 

### 📸 Screenshots

_No response_

### 📋 Logs

_No response_

### 🌐 Affected Browsers

_No response_

### 💻 Operating System

_No response_

### 🤝 Want to Contribute to the Fix?

Yes, I'd like to help fix this.

### ℹ️ Additional Information

_No response_

### PR DESCRIPTION
**Pull request by hcvdwerf**
…onment variables closes #147

### CODE REVIEW

**Comment by inderps** [comment]
@hcvdwerf Ok on prod yes once the build is created, you cannot change it because during build process, NEXTJS will replace `process.env. NEXT_PUBLIC_CKAN_URL` with the actual value.

So we must provide the actual value during the build process itself. To differentiate, we could have also named it `NEXT_BUILDTIME_PUBLIC_CKAN_URL` to imply that this env var is to be changed only during build time.

So my question is under what scenario, we would want to change the value after the build is done?

**Comment by hcvdwerf** [comment]
> @hcvdwerf Ok on prod yes once the build is created, you cannot change it because during build process, NEXTJS will replace `process.env. NEXT_PUBLIC_CKAN_URL` with the actual value.
> 
> So we must provide the actual value during the build process itself. To differentiate, we could have also named it `NEXT_BUILDTIME_PUBLIC_CKAN_URL` to imply that this env var is to be changed only during build time.
> 
> So my question is under what scenario, we would want to change the value after the build is done?

In scenarios where we are aiming for a flexible deployment process across different environments (such astest , acceptance, and production) without the need to create separate Docker images for each, environment variables are critical . They allow us to adjust configurations dynamically, post-build, to match the specific requirements of each environment.

One common reason for changing values like `NEXT_PUBLIC_CKAN_URL` after the build process is to accommodate the need for different API endpoints or service URLs in various environments. For instance, during development, you might connect to a staging API, but in production, you would need to connect to the live API. By utilizing environment variables effectively, we can switch these configurations seamlessly by simply changing the environment variable values at runtime, without necessitating a rebuild of the application.

**Review by inderps — APPROVED** [review]

File: src/config/serverConfig.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
 interface ServerConfig {
   ckanUrl: string;
   daamUrl: string;
 }
 
 const serverConfig: ServerConfig = {
-  ckanUrl: process.env.NEXT_PUBLIC_CKAN_URL || 'https://ckan-test.healthdata.nl',
+  ckanUrl: env('NEXT_PUBLIC_CKAN_URL') || 'https://ckan-test.healthdata.nl',
   daamUrl: process.env.DAAM_URL || 'http://localhost:8080',
```
Comment (brunopacheco1) [inline src/config/serverConfig.ts:13]: important: can you please also do it for `DAAM_URL`?

Comment (hcvdwerf) [reply src/config/serverConfig.ts]: check

**Review by brunopacheco1 — APPROVED** [review]

**Pull request merged**